### ISSUE TITLE
ClassCastException in org.springdoc.core.AbstractResponseBuilder#calculateSchema

### ISSUE THREAD

**Issue by andbed** [springdoc/springdoc-openapi#76]
I cannot use Springdoc in my project since it fails with exception:

```
java.lang.ClassCastException: sun.reflect.generics.reflectiveObjects.TypeVariableImpl cannot be cast to java.lang.Class
	at org.springdoc.core.AbstractResponseBuilder.calculateSchema(AbstractResponseBuilder.java:204)
	at org.springdoc.core.AbstractResponseBuilder.buildContent(AbstractResponseBuilder.java:184)
	at org.springdoc.core.AbstractResponseBuilder.buildApiResponses(AbstractResponseBuilder.java:239)
	at org.springdoc.core.AbstractResponseBuilder.computeResponse(AbstractResponseBuilder.java:155)
	at org.springdoc.core.AbstractResponseBuilder.build(AbstractResponseBuilder.java:62)
	at org.springdoc.api.AbstractOpenApiResource.calculatePath(AbstractOpenApiResource.java:124)
	at org.springdoc.api.OpenApiResource.getPaths(OpenApiResource.java:80)
	at org.springdoc.api.AbstractOpenApiResource.getOpenApi(AbstractOpenApiResource.java:71)
	at org.springdoc.api.OpenApiResource.openapiJson(OpenApiResource.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:104)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:892)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:797)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1039)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:942)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1005)
	... 45 more
```

The value of `returnType` is `T` in which is a generic type. It is the return type of generic `GET` method defined in an abstract controller.
I tried to use `@ApiResponses` to directly point to proper return type but it didn't help. Any advice?

**Comment by springdoc** [comment]
Hi, 

 Please add a sample of your REST Controller in order to reproduce the issue.

**Comment by springdoc** [comment]
A fix will be available on v1.1.36 of springdoc-openapi.
Please test it and give us your feedback. As we haven't sample code to reproduce your issue.
